app/code/Magento/CatalogCustomerGraphQl/Model/Resolver/PriceTiers.php

```diff
@@ -152,7 +155,7 @@ private function formatAndFilterTierPrices(
         array $tierPrices,
         string $currencyCode
     ): array {
-
+        $this->resetFormatAndFilterTierPrices();
```

Comment:
Why not simply cleaning the `formatAndFilterTierPrices`? Any concerns about that?
```suggestion
        $this->formatAndFilterTierPrices = [];
```

Reply:
Hi @eduard13 
Unfortunately this approach isn't working in this case 
We are losing some data
We will get tier price only for first item in this case 
**Example:**
<img width="923" alt="Screenshot 2021-03-03 at 15 51 11" src="https://user-images.githubusercontent.com/13517181/109815650-8b2e1d00-7c38-11eb-8d2d-f10d2df1aae6.png">
But actually  `banana-1` product has tier price

Reply:
But, I think you do the same in `resetFormatAndFilterTierPrices ` below, cleaning the `formatAndFilterTierPrices`. Am I wrong?

Reply:
Hi @eduard13 
Yes you are right. It was my mistake 🤦 
Fixed this one 
Thanks!